orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -114,3 +119,43 @@ func makeErr(errString string) *protos.RegisterResponse {
 	}
 	return errRes
 }
+
+// createGatewayDevice creates the gateway device model
+func createGatewayDevice(hwID *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) *models.GatewayDevice {
+	challengeKeyBase64 := strfmt.Base64(challengeKey.Key)
+	return &models.GatewayDevice{
+		HardwareID: hwID.Id,
+		Key:        &models.ChallengeKey{KeyType: challengeKey.KeyType.String(), Key: &challengeKeyBase64},
+	}
+}
+
+// updateGatewayDevice writes to the device information to the gateway entity
+func updateGatewayDevice(ctx context.Context, deviceInfo *protos.GatewayDeviceInfo, hwID *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error {
+	networkID := deviceInfo.NetworkId
+	gatewayID := deviceInfo.LogicalId
+
+	ent, err := configurator.LoadEntity(
+		ctx,
+		networkID, orc8r.MagmadGatewayType, gatewayID,
+		configurator.EntityLoadCriteria{},
+		serdes.Entity,
+	)
```

Comment:
I think there's a "load entity by physical ID" func

Reply:
I think the func `LoadEntityForPhysicalID` gets the entity by the physical ID, but since the gateway is not yet registered to a physical ID, idt it would work... unless there is some load entity by physical ID func u are referencing?

Reply:
Oh true you're right, I misread "gateway ID" as meaning "hardware ID" -- a +1 for why functions with large # of arguments is an anti-pattern 🙂